src/DotNetBridge/NativeDataView.cs

```diff
@@ -200,7 +140,9 @@ public NativeDataView(IHostEnvironment env, DataSourceBlock* pdata)
                 }
 
                 _columns = columns.ToArray();
-                Schema = Schema.Create(new SchemaImpl(_columns));
+                var schemaBuilder = new SchemaBuilder();
+                schemaBuilder.AddColumns(columns.Select(c => c.DetachedColumn));
+                Schema = schemaBuilder.GetSchema();
```

Comment:
>Schema [](start = 16, length = 6)

This object still holds name and type information stored in `_columns`, which means duplicated information. More specifically, `Schema[i]` holds everything stored in `_columns[i].DetachedColumn`. Is it possible to somehow deduplicate it?

Reply:
This is how I see it:
1) NativeDataView.Column class (and its 20 derived classes) handle reading/writing from native interop for different data types
2) NativeDataView.Schema is used for managed ML.NET side
3) NativeDataView.Column.DetachedColumn is replacing the SchemaImpl and contains duplicate info for type/name.

The only way to get rid of #3 is to redo the interfaces in #1 to not include type/name info. I don't see a big ROI for changing 20+ classes. I think having #3 is a good cost compromise.

---
In reply to: [250393432](https://github.com/Microsoft/NimbusML/pull/90#discussion_r250393432) [](ancestors = 250393432)

Reply:
It's ok to keep it as long as we document this behavior in the doc strings of `_columns` and `Schema`. We can say `Schema[i]` holds the same `Name` and `Type` stored in `_columns[i].DetachedColumn`.

---
In reply to: [250419270](https://github.com/Microsoft/NimbusML/pull/90#discussion_r250419270) [](ancestors = 250419270,250393432)

Reply:
Added the notes.

---
In reply to: [250421565](https://github.com/Microsoft/NimbusML/pull/90#discussion_r250421565) [](ancestors = 250421565,250419270,250393432)